pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingData.java

```diff
@@ -130,8 +132,97 @@ public long skip(final long n) {
             long numSkipped = in.skip(clamped);
             position += numSkipped;
             return numSkipped;
-        } catch (IOException e) {
+        } catch (final IOException e) {
+            throw new DataAccessException(e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public long readBytes(@NonNull final byte[] dst, final int offset, final int maxLength) {
+        if (maxLength < 0) {
+            throw new IllegalArgumentException("Negative maxLength not allowed");
+        }
+        if (!hasRemaining()) {
+            return 0;
+        }
+        try {
+            int bytesRead = in.read(dst, offset, maxLength);
```

Comment:
If we use `in.readNBytes` here (same parameters) then we don't need the if loop below as the read will block until the bytes are read or EOS is reached. [Docs here.](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/InputStream.html#readNBytes(byte%5B%5D,int,int))
>This method blocks until the requested number of bytes has been read, end of stream is detected, or an exception is thrown.

It is worth noting that the default implementation of `read` actually behaves the same, but `readNBytes` is _specified_ to behave in this manner.
>The default implementation of this method blocks until the requested amount of input data len has been read, end of file is detected, or an exception is thrown.

Reply:
Switched to use `readNBytes()` here